### PR TITLE
Phase-2 L1: code for L1 demonstrators

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This is the 2nd PR to port all L1T Phase-2 developments to cms-sw. This PR targets the demonstrator tools for the L1 boars. It includes several PRs made locally, the details of which can be found in the twiki https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideL1TPhase2Instructions and in the cms-l1t-offline branch.

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
> A new Pull Request was created by @cecilecaillol for master.
> 
> It involves the following packages:
> 
>     * L1Trigger/DemonstratorTools (****)
> 
> 
> The following packages do not have a category, yet:
> 
> L1Trigger/DemonstratorTools Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category
> 
> @cmsbuild can you please review it and eventually sign? Thanks. @Martin-Grunewald, @missirol this is something you requested to watch as well. @perrotta, @dpiparo, @qliphy you are the release manager for this.
> 
> cms-bot commands are listed [here](http://cms-sw.github.io/cms-bot-cmssw-cmds.html)

The new package was added to categories_map but it is not reflected in the flag here (new-package-pending)

**Comment by cecilecaillol** [comment]
+l1

**Comment by cecilecaillol** [comment]
@perrotta, @dpiparo, @qliphy Could you please review this PR? Our next PRs rely on this code

File: L1Trigger/DemonstratorTools/plugins/GTTFileReader.cc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  iEvent.put(std::make_unique<l1t::VertexWordCollection>(vertices));
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void GTTFileReader::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.setUnknown();
```
Comment (perrotta) [inline L1Trigger/DemonstratorTools/plugins/GTTFileReader.cc:102]: You have only two parameters: please allow real parameter description and validation here

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Pull request merged**